core/src/main/scala/kafka/cluster/Partition.scala

```diff
@@ -235,10 +233,20 @@ class Partition(val topic: String,
   def updateReplicaLogReadResult(replicaId: Int, logReadResult: LogReadResult) {
     getReplica(replicaId) match {
       case Some(replica) =>
+        // No need to calculate low watermark if there is no delayed DeleteRecordsRequest
+        val oldLeaderLW = if (replicaManager.delayedDeleteRecordsPurgatory.delayed() > 0) lowWatermarkIfLeader else -1L
         replica.updateLogReadResult(logReadResult)
+        val newLeaderLW = if (replicaManager.delayedDeleteRecordsPurgatory.delayed() > 0) lowWatermarkIfLeader else -1L
```

Comment:
If there is only one live replica for this partition, then LW will increase immediately to the specified offset and the response can be sent to the client. Otherwise, say there is at least one live follower for this partition, LW will not increase now. LW will increase in the next updateReplicaLogReadResult() call, and the response can be sent back then. Does this work?